vowpalwabbit/oaa.cc

```diff
@@ -89,19 +91,23 @@ void predict_or_learn(oaa& o, LEARNER::base_learner& base, example& ec)
 
   if (is_probabilities)
   { float sum_prob = 0;
-    ec.pred.probs = calloc_or_throw<float>(o.k);
+    float* probs = calloc_or_throw<float>(o.k);
```

Comment:
This is mainly to avoid: ec.pred.probs. You rather have it spread all over (ec.pred.probs[i] = prob;)?

float* probs = calloc_or_throw<float>(o.k);
    for (uint32_t i=0; i<o.k; i++)
    { // probability of class (i+1) = logistic_link_function(raw_prediction)
      float prob = 1.f / (1.f + exp(- o.pred[i].scalar));
      probs[i] = prob;
      sum_prob += prob;
    }
    // make sure that the probabilities sum up (exactly) to one
    for (uint32_t i=0; i<o.k; i++)
      probs[i] /= sum_prob;

       ec.pred.probs = probs;

From: John [mailto:notifications@github.com]
Sent: Friday, August 12, 2016 3:25 PM
To: JohnLangford/vowpal_wabbit <vowpal_wabbit@noreply.github.com>
Cc: Markus Cozowicz <marcozo@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [JohnLangford/vowpal_wabbit] ADF offline eval / quiet;-t improvements (#1069)


In vowpalwabbit/oaa.cc<https://github.com/JohnLangford/vowpal_wabbit/pull/1069#discussion_r74647241>:

> @@ -89,19 +91,23 @@ void predict_or_learn(oaa& o, LEARNER::base_learner& base, example& ec)

>

>    if (is_probabilities)

>    { float sum_prob = 0;

> -    ec.pred.probs = calloc_or_throw<float>(o.k);

> +    float* probs = calloc_or_throw<float>(o.k);

Can we avoid a spurious allocation?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/JohnLangford/vowpal_wabbit/pull/1069/files/a6a98c1119690f665453bef089a171e3ee294820#r74647241>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AJrCTur-qAfEWvXkIxxCyM9ifY8wkMuMks5qfMh2gaJpZM4JhZyM>.